cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js

```diff
@@ -117,11 +117,63 @@ function getPaymentMethods(cart) {
     if (Site.getCurrent().getCustomPreferenceValue("Adyen_directoryLookup")) {
         var	getPaymentMethods = require('int_adyen_overlay/cartridge/scripts/adyenGetPaymentMethods');
         var paymentMethods = getPaymentMethods.getMethods(cart.object).paymentMethods;
-        return paymentMethods.filter(function (method) { return method.type != "scheme"; })
+        return paymentMethods.filter(function (method) { 
+        	return isPaymentMethodTypeNotFilteredOut(method.type); 
+        });
     }
     return {};
 }
 
+/**
+ * Checks if payment method is not filtered out
+ */
+function isPaymentMethodTypeNotFilteredOut(methodType) {
+	if (!isPaymentMethodTypeScheme(methodType) && 
+		!isPaymentMethodTypeBcmcMobileQR(methodType) &&
+		!isPaymentMethodTypeWechatpayExceptWeb(methodType)
+	) {
+		return true;
+	}
+	
+	return false;
+}
```

Comment:
I think we can simplify this a bit more by adding a common method for filtered types.

Reply:
Agree